### ISSUE TITLE
User foreign key getting error when try to delete the user from the database.

### ISSUE THREAD

**Issue by c0d33py** [Corvia/django-tenant-users#330]
## Settings

```python
TENANT_APPS = (
    # Django Apps
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'tenant_users.permissions',
    # Project Apps
    'core',
)
```
## core/models.py

```python
class Category(models.Model):
    name = models.CharField(max_length=100)
    description = models.TextField(blank=True, null=True)

    created_by = models.ForeignKey(
        settings.AUTH_USER_MODEL,
        on_delete=models.CASCADE,
        related_name='%(class)s_created_by',
        null=True, blank=True)
```

## Actual Behavior

``` 
Traceback(most recent call last):
     return self.cursor.execute(sql, params)
 django.db.utils.ProgrammingError: relation "core_category" does not exist
 LINE 1: ...er"."is_verified", "users_tenantuser"."name" FROM "core_cate...
```
## Expected Behavior

 I want to delete the user from the tenant database. When i try to delete the
 user from the tenant database, it gives me the error above.
 If i remove the `created_by` field from the `Category` model, it works fine.
 The foreign key is pointing to the `users.TenantUser` model.

## Steps to Reproduce

1. Create a new tenant
2. Create a new user in the tenant
3. Create a new category in the tenant
4. Create another new user in the tenant and try to delete it
5. It gives the error above when you try to delete the user

## Additional Context
I am using django-tenant-users for multi-tenancy. I am using django-tenants
This is the [example](https://github.com/c0d33py/user-tenant) app which i used to create the public tenant.
https://github.com/c0d33py/user-tenant

## Your Environment
- Django == 4.1.2
- django-tenant-users == 1.1.0
- django-tenants == 3.4.5
- psycopg2-binary == 2.9.4

**Comment by Dresdn** [comment]
Hi @c0d33py - I'm getting a 404 when trying to access your demo app.

I'm curious how you have everything set up as you don't "delete the user from the tenant database" at all, but rather assign/remove permissions to a tenant for a given user.

If you have everything setup right with your users in the `SHARED_APPS` and wanting the `Category` deletion to happen when you delete a User, then that's probably more a question for `django-tenants` than this project.

**Comment by c0d33py** [comment]
Please take a look. Now you have the access to the `Example` app repository.

**Comment by Dresdn** [comment]
This is definitely within the scope of `django-tenants` as this has to do with creating ForeignKey relationships between a Shared App and Public App. Doing a quick search, I do see this issue, which is exactly what you're seeing: [django-tenants/django-tenant#592](https://github.com/django-tenants/django-tenants/issues/592)

The alternative would be to create your own Manager command to delete a user that would go through all tenants, find any `Category` that was `created_by` the given user, delete those first, and then delete the `TenantUser` object. For your `Category.created_by` field, I'd probably look at changing the `on_delete` parameter, but not too sure it'd help any due to the underlying issue.

Hope this helps.

**Comment by Dresdn** [comment]
Thinking about the `wontfix` label, if there is a way to incorporate a "fix" into either the documentation or through some other means, I do think it would be a great enhancement to add to the project. That is unless the fix should live in `django-tenants`. I haven't looked through the issues more, so please update if you find anything interesting.

**Comment by mihalikv** [comment]
@Dresdn this is still an issue, I am currently using django tenants with wagtail, with following configuration:
```python
SHARED_APPS = (
    'django_tenants',

    ....

    'dynamic_preferences',
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    'django.contrib.sites',
    'django.contrib.sitemaps',
    'django_advanced_password_validation',
)

TENANT_APPS = (
    # - - START Wagtail requirements - -
    "wagtail_localize",
    "wagtail_localize.locales",
    "wagtail.contrib.forms",
    'wagtail.contrib.redirects',
    'wagtail.embeds',
    'wagtail.sites',
    'wagtail.users',
    'wagtail.snippets',
    'wagtail.documents',
    'wagtail.images',
    'wagtail.search',
    'wagtail.admin',
    'wagtail',

    'modelcluster',
    'taggit',

    'wagtail.api.v2',
    'wagtail_color_panel',
    'project.cms'
    # - - END - -
)
```

And there is an FK to User in wagtail https://github.com/wagtail/wagtail-localize/blob/main/wagtail_localize/models.py#L1636. So I am not able to delete the user. Because django look for StringTranslation in public schema. 

1. I am not able to override on_delete behaviour because, it's a library.
2. I don't have custom User model so I need to probably monekypatch .delete() to override this behaviour.

Do you have some other advice how to overcome this issue ?

Thank you.

**Comment by Dresdn** [comment]
Looking through the `django-tenants` project, there's a few issues in there around relating models between the `Public` and `Tenant` schema:

https://github.com/bernardopires/django-tenant-schemas/issues/420
https://github.com/django-tenants/django-tenants/issues/562
https://github.com/django-tenants/django-tenants/issues/592

So you can see this is a fairly common scenario. Looking through one of the issues, I recalled this comment which I think sums up exactly what's going on, and what you need to do:

https://github.com/bernardopires/django-tenant-schemas/issues/420#issuecomment-301223581

My suggestion would be to start with looking at how Django itself handles cascade deletes (a la https://github.com/django/django/blob/main/django/db/models/deletion.py#L96)

Basically, when wanting to delete the `User` object, your custom method will need to get all the objects that would be impacted by the deletion, and then call the deletion of those objects within each tenant schema, and then delete the User object itself.

Now that I say all that, perhaps it should live in this project. If you want to take a stab at writing something that could be agnostic, more than happy to help collaborate and merge it into the app.
